### PR TITLE
Fix links for book editions.

### PR DESCRIPTION
**Pull request by ehuss**
Now that there is one book to rule them all, some of the links needed updating.

The prose needed significant adjustment to accommodate the most recent changes.
I did my best to capture what is going on, but I am not confident it is correct.

### CODE REVIEW

File: src/rust-2018/documentation/new-editions-of-the-book.md [hunk 1]
```diff
@@ -15,23 +17,19 @@ details were nailed down for the 1.0 release. It didn't do a fantastic job of
...
-The final version was shipped with Rust 1.26. The new edition is a complete
+The final version was shipped with Rust 1.26. The second edition is a complete
 re-write from the ground up, using the last two years of knowledge we’ve
-gained from teaching people Rust. You’ll find brand-new explanations for a
-lot of Rust’s core concepts, new projects to build, and all kinds of other
-good stuff. Please [check it
-out](https://doc.rust-lang.org/book/second-edition/index.html) and let us
+gained from teaching people Rust.
+
+You can purchase [a dead-tree version of the second edition from No Starch
```
Comment (Centril) [inline src/rust-2018/documentation/new-editions-of-the-book.md]: Can we use a less derogatory form of "book in paper form"?

File: src/rust-2018/documentation/new-editions-of-the-book.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 ![Minimum Rust version: 1.28](https://img.shields.io/badge/Minimum%20Rust%20Version-1.28-red.svg) for drafts of the 2018 edition
```
Comment (siddharthasahu) [inline src/rust-2018/documentation/new-editions-of-the-book.md]: Why are the rust versions noted for **draft** versions of the books? Can they not just be removed?

Comment (ehuss) [reply src/rust-2018/documentation/new-editions-of-the-book.md]: Sounds reasonable to me.

Comment (siddharthasahu) [reply src/rust-2018/documentation/new-editions-of-the-book.md]: Thanks @ehuss!

**Pull request merged**